mongo/integration/unified/entity.go

```diff
@@ -401,6 +429,178 @@ func (em *EntityMap) addDatabaseEntity(entityOptions *entityOptions) error {
 	return nil
 }
 
+var (
+	envAwsAccessKeyID         = os.Getenv("AWS_ACCESS_KEY_ID")
+	envAwsSecretAccessKey     = os.Getenv("AWS_SECRET_ACCESS_KEY")
+	envAwsTempAccessKeyID     = os.Getenv("CSFLE_AWS_TEMP_ACCESS_KEY_ID")
+	envAwsTempSecretAccessKey = os.Getenv("CSFLE_AWS_TEMP_SECRET_ACCESS_KEY")
+	envAwsTempSessionToken    = os.Getenv("CSFLE_AWS_TEMP_SESSION_TOKEN")
+	envAzureTenantID          = os.Getenv("AZURE_TENANT_ID")
+	envAzureClientID          = os.Getenv("AZURE_CLIENT_ID")
+	envAzureClientSecret      = os.Getenv("AZURE_CLIENT_SECRET")
+	envGcpEmail               = os.Getenv("GCP_EMAIL")
+	envGcpPrivateKey          = os.Getenv("GCP_PRIVATE_KEY")
+)
+
+// getKmsCredential processes a value of an input KMS provider credential.
+// An empty document returns from the environment.
+// A string is returned as-is.
+func getKmsCredential(kmsDocument bson.Raw, credentialName string, defaultValue string) (*string, error) {
```

Comment:
Optional: If all `defaultValue`s come from environment variables, consider replacing the `defaultValue` parameter with an environment variable name parameter and perform the env var lookup in `getKmsCredential`. That would allow more specific messaging about any missing environment variables rather than the less specific "Please set CSFLE environment variables" message.

Reply:
Great idea. Some credentials have default values and no environment variable. Updated to separate the `defaultValue` from the environment variable, and print the environment variable in the error message.